File: problems/001_leetcode_id_509_fibonacci_number/python/approach_5.py

#  Method 5: Binet's Formula (Closed-form expression)

def fib(n):
        sqrt5 = 5 ** 0.5
        fibN = ((1 + sqrt5) / 2) ** n - ((1 - sqrt5) / 2) ** n
        return round(fibN / sqrt5)

input = 4
print(fib(input)); # Output: 3

File: problems/002_leetcode_id_1838_frequency_of_the_most_frequent_element/javascript/approach_5.js

// Method 5: Binet's Formula (Closed-form expression)

var fib = function(n) {
    const sqrt5 = Math.sqrt(5);
    const fibN = Math.pow((1 + sqrt5) / 2, n) - Math.pow((1 - sqrt5) / 2, n);
    return Math.round(fibN / sqrt5);
};

input = 4;
console.log(fib(input)); // Output: 3